### PR TITLE
kafka(ticdc): make column selector can work normally to filter out columns

### LINKED ISSUE #9967: column selector works normally filter out columns

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9967 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: cdc/sink/dmlsink/mq/transformer/columnselector/column_selector.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		event.PreColumns[idx] = nil
+	}
+
+	return nil
+}
+
+// ColumnSelector manages an array of selectors, the first selector match the given
+// event is used to select out columns.
+type ColumnSelector struct {
+	selectors []*selector
```
Comment (nongfushanquan) [inline cdc/sink/dmlsink/mq/transformer/columnselector/column_selector.go:96]: why does it need lots of selector?

**Review by nongfushanquan — COMMENTED** [review]
how about the situation with the Avro protocol ?

File: cdc/sink/dmlsink/mq/transformer/columnselector/column_selector.go (hunk 1 above)
Comment (3AceShowHand) [reply cdc/sink/dmlsink/mq/transformer/columnselector/column_selector.go:96]: Each `selector` matches a subset of tables.

**Review by sdojjy — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**